sentinel-core/src/main/java/com/alibaba/csp/sentinel/slots/clusterbuilder/ClusterBuilderSlot.java

```diff
@@ -81,7 +81,7 @@ public void entry(Context context, ResourceWrapper resourceWrapper, DefaultNode
                 if (clusterNode == null) {
                     // Create the cluster node.
                     clusterNode = Env.nodeBuilder.buildClusterNode();
-                    HashMap<ResourceWrapper, ClusterNode> newMap = new HashMap<ResourceWrapper, ClusterNode>(16);
+                    HashMap<ResourceWrapper, ClusterNode> newMap = new HashMap<ResourceWrapper, ClusterNode>(Math.max(clusterNodeMap.size(), 16));
```

Comment:
First, we can use type inference instead of explicit type parameter here:

```Java
new HashMap<>(...);
```

Second, the initial size should be the max of `(int) clusterNodeMap.size() / 0.75 + 1` and 16.